packages/components/src/components/TaskRunDetails/TaskRunDetails.js

```diff
@@ -54,7 +54,16 @@ function getDescriptions(array) {
   }, {});
 }
 
-const TaskRunDetails = ({ onViewChange, pod, task, taskRun, view }) => {
+const noop = /* istanbul ignore next */ () => {};
+const emptyObj = {};
+
+const TaskRunDetails = ({
+  onViewChange = noop,
+  pod,
+  task = emptyObj,
+  taskRun = emptyObj,
```

Comment:
Any chance that setting `task` and `taskRun` to the same initial object could result in cross-contamination when values are added to either object?

Reply:
It shouldn't as the props shouldn't be modified by the component but it's a risk, and would be difficult to debug in future if we ran into a problem caused by it.

I've updated all of these to define a `defaults` object (similar to the old `defaultProps`) assigning the value for each affected prop, and reference that in the params instead.

So for example this one becomes:
```js
task = defaults.task,
taskRun = defaults.taskRun,
```